src/DIRAC/Core/Base/AgentModule.py

```diff
@@ -154,18 +154,16 @@ def __init__(self, agentName, loadName, baseAgentName=False, properties={}):
         self.__initialized = False
 
     def __getCodeInfo(self):
-        versionVar = "__RCSID__"
         docVar = "__doc__"
         try:
-            self.__agentModule = __import__(self.__class__.__module__, globals(), locals(), versionVar)
+            self.__agentModule = __import__(self.__class__.__module__, globals(), locals(), "__doc__")
         except Exception as excp:
             self.log.exception("Cannot load agent module", lException=excp)
-        for prop in ((versionVar, "version"), (docVar, "description")):
-            try:
-                self.__codeProperties[prop[1]] = getattr(self.__agentModule, prop[0])
-            except Exception:
-                self.log.error("Missing property", prop[0])
-                self.__codeProperties[prop[1]] = "unset"
+        try:
```

Comment:
How about:
```suggestion
        self.__codeProperties["version"] = importlib.metadata.version(inspect.getmodule(self).__package__.split(".")[0])
        try:
```
or possibly:
```suggestion
        try:
            self.__codeProperties["version"] = importlib.metadata.version(inspect.getmodule(self).__package__.split(".")[0])
        except Exception:
            self.log.exception(f"Failed to find version for {self!r}")
            self.__codeProperties["version"] = "unset"
        try:
```

(I feel like there should be a better way than `.__package__.split(".")[0]` but I can't think of one...)